src/sinks/aws_kinesis_streams/integration_tests.rs

```diff
@@ -0,0 +1,130 @@
+#![cfg(feature = "aws-kinesis-streams-integration-tests")]
+#![cfg(test)]
+
+use super::*;
+use crate::config::SinkConfig;
+use crate::sinks::util::encoding::StandardEncodings;
+use crate::sinks::util::{BatchConfig, Compression};
+use crate::{
+    config::SinkContext,
+    rusoto::RegionOrEndpoint,
+    test_util::{random_lines_with_stream, random_string},
+};
+use rusoto_core::Region;
+use rusoto_kinesis::{Kinesis, KinesisClient};
+use std::sync::Arc;
+use tokio::time::{sleep, Duration};
+
+#[tokio::test]
+async fn kinesis_put_records() {
+    let stream = gen_stream();
+
+    let region = Region::Custom {
+        name: "localstack".into(),
+        endpoint: "http://localhost:4566".into(),
+    };
+
+    ensure_stream(region.clone(), stream.clone()).await;
+
+    let config = KinesisSinkConfig {
+        stream_name: stream.clone(),
+        partition_key_field: None,
+        region: RegionOrEndpoint::with_endpoint("http://localhost:4566".into()),
+        encoding: StandardEncodings::Text.into(),
+        compression: Compression::None,
+        batch: BatchConfig {
+            max_events: Some(2),
+            ..Default::default()
+        },
+        request: Default::default(),
+        assume_role: None,
+        auth: Default::default(),
+    };
+
+    let cx = SinkContext::new_test();
+
+    let sink = config.build(cx).await.unwrap().0;
+
+    let timestamp = chrono::Utc::now().timestamp_millis();
+
+    let (mut input_lines, events) = random_lines_with_stream(100, 11, None);
+
+    let _ = sink.run(events).await.unwrap();
+
+    sleep(Duration::from_secs(1)).await;
```

Comment:
Could you add component tests here:
```suggestion
    components::init_test();
    let _ = sink.run(events).await.unwrap();

    sleep(Duration::from_secs(1)).await;
    components::SINK_TESTS.assert(&["region"]);
```

It's probably worth adding that array of tags to `src/test_utils/components.rs` as it will be used in other AWS components.